### PR TITLE
Ports: Update 79 ports

### PR DESCRIPTION
**Pull request by EWouters**
All of the formatting changes have been combined into a single commit. These have been tested on the GNU build on an M1. Most of them work on the Clang build as well, but not all of them.

### CODE REVIEW

File: Ports/SDL2_image/package.sh [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 port='SDL2_image'
-useconfigure='true'
 version='2.6.2'
-depends=("SDL2" "libpng" "libjpeg" "libtiff")
+useconfigure='true'
+depends=(
+    'SDL2'
+    'libpng'
+    'libjpeg'
+    'libtiff'
```
Comment (timschumi) [inline Ports/SDL2_image/package.sh:9]: Sort alphabetically.

File: Ports/SDL2_mixer/package.sh [hunk 2]
```diff
@@ -5,23 +5,20 @@ useconfigure='true'
...
-
-post_configure() {
-    unset LIBS
-}
+depends=(
+    'libmodplug'
+    'libmpg123'
+    'libvorbis'
+    'SDL2'
+    'timidity')
```
Comment (timschumi) [inline Ports/SDL2_mixer/package.sh:13]:
```suggestion
    'timidity'
)
```

File: Ports/SDL2_ttf/package.sh [hunk 3]
```diff
@@ -5,15 +5,16 @@ useconfigure='true'
 files=(
     "https://github.com/libsdl-org/SDL_ttf/releases/download/release-${version}/SDL2_ttf-${version}.tar.gz 78cdad51f3cc3ada6932b1bb6e914b33798ab970a1e817763f22ddbfd97d0c57"
 )
-depends=("SDL2" "freetype")
+depends=(
+    'SDL2'
+    'freetype'
```
Comment (timschumi) [inline Ports/SDL2_ttf/package.sh:10]: Sort alphabetically.

File: Ports/fontconfig/package.sh [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
-port=fontconfig
-version=2.14.0
-useconfigure="true"
-use_fresh_config_sub="true"
-depends=("libxml2" "freetype")
+port='fontconfig'
+version='2.14.0'
+depends=(
+    'libxml2'
+    'freetype'
```
Comment (timschumi) [inline Ports/fontconfig/package.sh:6]: Sort alphabetically.

File: Ports/git/package.sh [hunk 5]
```diff
@@ -5,11 +5,19 @@ files=(
...
-depends=("zlib" "curl")
+configopts=(
+    "--target=${SERENITY_ARCH}-pc-serenity"
+    "--with-lib=${SERENITY_INSTALL_ROOT}/usr/local"
+    'CFLAGS=-DNO_IPV6'
+    "LDFLAGS=-L${SERENITY_INSTALL_ROOT}/usr/local/lib"
+)
+depends=(
+    'zlib'
+    'curl'
```
Comment (timschumi) [inline Ports/git/package.sh:16]: Sort alphabetically.

File: Ports/gnupg/package.sh [hunk 6]
```diff
@@ -1,19 +1,30 @@
...
+config_sub_paths=(
+    'build-aux/config.sub'
+)
+configopts=(
+    "--with-libgpg-error-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+    "--with-libgcrypt-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+    "--with-libassuan-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+    "--with-ntbtls-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+    "--with-npth-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+    '--disable-dirmngr')
```
Comment (timschumi) [inline Ports/gnupg/package.sh:16]:
```suggestion
    '--disable-dirmngr'
)
```

File: Ports/gnupg/package.sh [hunk 7]
```diff
@@ -1,19 +1,30 @@
...
 )
-depends=("libiconv" "libgpg-error" "libgcrypt" "libksba" "libassuan" "npth" "ntbtls")
+depends=(
+    'libiconv'
+    'libgpg-error'
+    'libgcrypt'
+    'libksba'
+    'libassuan'
+    'npth'
+    'ntbtls'
```
Comment (timschumi) [inline Ports/gnupg/package.sh:27]: Sort alphabetically.

File: Ports/imagemagick/package.sh [hunk 8]
```diff
@@ -2,9 +2,15 @@
...
 )
-configopts=("--with-sysroot=${SERENITY_INSTALL_ROOT}")
-depends=("libpng" "libtiff" "libjpeg")
+configopts=(
+    "--with-sysroot=${SERENITY_INSTALL_ROOT}"
+)
+depends=(
+    'libpng'
+    'libtiff'
+    'libjpeg'
```
Comment (timschumi) [inline Ports/imagemagick/package.sh:15]: Sort alphabetically.

File: Ports/libarchive/package.sh [hunk 9]
```diff
@@ -1,13 +1,20 @@
...
+configopts=(
+    '--without-xml2'
+)
 files=(
     "https://libarchive.org/downloads/libarchive-${version}.tar.gz c676146577d989189940f1959d9e3980d28513d74eedfbc6b7f15ea45fe54ee2"
 )
-depends=("zlib" "pcre")
+depends=(
+    'zlib'
+    'pcre'
```
Comment (timschumi) [inline Ports/libarchive/package.sh:17]: Sort alphabetically.

File: Ports/libicu/package.sh [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
+workdir='icu/source'
 files=(
     "https://github.com/unicode-org/icu/releases/download/release-${version//./-}/icu4c-${version//./_}-src.tgz 4cba7b7acd1d3c42c44bb0c14be6637098c7faf2b330ce876bc5f3b915d09745"
 )
 
 configure() {
     host_env
     run mkdir -p ../host-build
-    run sh -c "cd ../host-build && ../source/configure && make ${makeopts[@]}"
+    run sh -c "cd ../host-build && ../source/configure && make ${makeopts[*]}"
```
Comment (timschumi) [inline Ports/libicu/package.sh:14]: Does this actually make any difference in practice?

File: Ports/libtiff/package.sh [hunk 11]
```diff
@@ -7,9 +7,13 @@ files=(
...
+    '--prefix=/usr/local'
+    '--disable-static'
+    '--enable-shared'
 )
 workdir="tiff-${version}"
-depends=("libjpeg" "zstd" "xz")
+depends=(
+    'libjpeg'
+    'zstd'
+    'xz'
```
Comment (timschumi) [inline Ports/libtiff/package.sh:18]: Sort alphabetically.

File: Ports/ntbtls/package.sh [hunk 12]
```diff
@@ -1,10 +1,17 @@
...
+useconfigure='true'
+use_fresh_config_sub='true'
+config_sub_paths=(
+    'build-aux/config.sub'
+)
+depends=(
+    'libgpg-error'
+    'libksba'
+    'libgcrypt'
+    'zlib'
```
Comment (timschumi) [inline Ports/ntbtls/package.sh:13]: Sort alphabetically.

File: Ports/tig/package.sh [hunk 13]
```diff
@@ -1,8 +1,13 @@
...
+useconfigure='true'
 files=(
     "https://github.com/jonas/tig/releases/download/tig-${version}/tig-${version}.tar.gz 24ba2c8beae889e6002ea7ced0e29851dee57c27fde8480fb9c64d5eb8765313"
 )
-depends=("ncurses" "pcre" "readline" "libiconv")
+depends=(
+    'ncurses'
+    'pcre'
+    'readline'
+    'libiconv'
```
Comment (timschumi) [inline Ports/tig/package.sh:12]: Sort alphabetically.

File: Ports/xz/package.sh [hunk 14]
```diff
@@ -1,11 +1,19 @@
...
-version=5.2.5
-useconfigure=true
-configopts=("--disable-static" "--enable-shared")
-use_fresh_config_sub=true
-config_sub_paths=("build-aux/config.sub")
+port='xz'
+version='5.2.5'
+depends=(
+    'zlib'
+    'libiconv'
```
Comment (timschumi) [inline Ports/xz/package.sh:6]: Sort alphabetically.

**Review by timschumi — CHANGES_REQUESTED** [review]
Took a preliminary look through the first commit and left some (probably non-exhaustive) feedback. The rest I'll probably be unable to test in some time.

If you want to get the first commit in early please put it in a separate PR.

Also, if you don't mind, please drop the changes where you move configure/make options from a custom command into the respective array. We've been discussing whether to do the reverse, and I have a local patchset somewhere that actively puts that into action.

**Comment by BertalanD** [comment]
The binutils version (and patch) is synced with the toolchain. Could you please leave out that change from this PR. I'm planning more involved changes than just bumping the version.

File: Ports/libicu/package.sh (hunk 10 above)
Comment (EWouters) [reply Ports/libicu/package.sh:14]: This removes a shellcheck warning: `Argument mixes string and array. Use * or separate argument`. Shellcheck [SC2145](https://www.shellcheck.net/wiki/SC2145).

File: Ports/SDL2_mixer/package.sh (hunk 2 above)
Comment (EWouters) [reply Ports/SDL2_mixer/package.sh]: Always some typos..

**Comment by EWouters** [comment]
I've removed binutils and got the initial comments sorted. I hope to merge this as one PR instead of splitting the formatting out of it. Most of these updates are very straightforward. I'll update some ports that are more complicated in a separate PR.

**Comment by EWouters** [comment]
This may be useful when testing:
```bash
ports=(bdwgc boost bzip3 ca-certificates ccache composer coreutils curl diffutils double-conversion e2fsprogs ed file flac fontconfig freetype gawk gettext git glib gmp gnupg grep gzip imagemagick isl jdupes kakoune less libarchive libassuan libatomic_ops libexpat libffi libgcrypt libgpg-error libicu libksba libmpg123 libopenal libpng libtiff libunistring libuuid libxml2 libzip links lua mawk mbedtls mpc mpfr ncurses ninja ntbtls ObjFW openrct2 openssl pcre2 php pixman pkgconf pt2-clone qoi r0 readline rsync sdl12-compat SDL2_image SDL2_mixer SDL2_ttf sqlite stress-ng tig vttest wget x264 xz zstd)
for i in ${ports} ; do bash -c "cd ${i} && ./package.sh" || break; done
```

**Comment by EWouters** [comment]
I nuked my checkout and build all the ports again with both toolchains. This time e2fsporgs failed to build on GNU, so I included a fix, and they now all build again on GNU. There were a few more ports I fixed for the Clang build. The ports that still don't build with Clang on my system (M1) are bdwgc, kakoune, links, SDL2_ttf, coreutils and php. I'm not able to check SDL2_image, sdl12-compat and SDL2_mixer because they depend on libmodplug, which doesn't build, and composer which depends on php.

**Review by gmta — APPROVED** [review]

**Comment by gmta** [comment]
Thanks @EWouters ! :^)

**Pull request merged**